pkg/helmexec/runner.go

```diff
@@ -34,7 +38,87 @@ func (shell ShellRunner) Execute(cmd string, args []string, env map[string]strin
 	preparedCmd := exec.Command(cmd, args...)
 	preparedCmd.Dir = shell.Dir
 	preparedCmd.Env = mergeEnv(os.Environ(), env)
-	return Output(preparedCmd)
+	return shell.OutputInLine(preparedCmd)
+}
+
+func (shell ShellRunner) OutputInLine(c *exec.Cmd) ([]byte, error) {
+	var errorStream string
+	var outputStream string
+	var combinedStream string
+
+	stdout, err := c.StdoutPipe()
+	if err != nil {
+		return nil, errors.New("exec: Stdout already set")
+	}
+	stderr, err := c.StderrPipe()
+	if err != nil {
+		return nil, errors.New("exec: Stderr already set")
+	}
+
+	err = c.Start()
+	if err != nil {
+		panic(fmt.Sprintf("unexpected error: %v", err))
+	}
+
+	combined := make(chan string, 2)
+
+	var wg sync.WaitGroup
+
+	scannerStdout := bufio.NewScanner(stdout)
+	wg.Add(1)
+	go func() {
+		for scannerStdout.Scan() {
+			text := scannerStdout.Text()
+			if strings.TrimSpace(text) != "" {
+				outputStream += text + "\n"
+				combined <- text
+			}
+		}
+		wg.Done()
+	}()
+
+	scannerStderr := bufio.NewScanner(stderr)
+	wg.Add(1)
+	go func() {
+		for scannerStderr.Scan() {
+			text := scannerStderr.Text()
+			if strings.TrimSpace(text) != "" {
+				errorStream += text + "\n"
+				combined <- text
+			}
+		}
+		wg.Done()
+	}()
+	go func() {
+		wg.Wait()
+		close(combined)
+	}()
+
+	for t := range combined {
```

Comment:
I think this for-range loop needs to be within a dedicated go routine. Otherwise doesn't it block indefinitely?

`close(combined)` is required to finish this block, which requires the `scannerStdout` and `scannerStderr` to close, but to close those two scanners `c.Wait` needs to be run concurrently.

Reply:
Hey thanks for the comments, no it doesn't block indefinitely but reads the output from the combined channel as its written to by the go routines that handle stdout and stderr..

I've tested the build binary on a few my local helmfile deployments and it's working well

Reply:
My fault! I just saw `err = c.Start()` in L58 which invalidates my comment. Thanks for confirming